frame.go

```diff
@@ -402,9 +402,11 @@ type framer struct {
 	wbuf []byte
 
 	customPayload map[string][]byte
+
+	flagLWT int
 }
 
-func newFramer(r io.Reader, w io.Writer, compressor Compressor, version byte) *framer {
+func newFramer(r io.Reader, w io.Writer, compressor Compressor, version byte, cqlProtoExts []cqlProtocolExtension) *framer {
```

Comment:
I do not like the fact we change framer constructor signature. 
Why we want to conditionally change framer?
If we really do want to do that there are better ways of implementing that.

Reply:
> I do not like the fact we change framer constructor signature.
> Why we want to conditionally change framer?
> If we really do want to do that there are better ways of implementing that.

Can you please expand on this one? Why changing the signature is bad?
I can't yet think of another way to customize serialization/deserialization
procedure based on which CQL protocol extensions we use.

It would be very nice if you could point me to some other way.

Reply:
One way would be to just move the check out of the constructor and apply it after the initial initialization perhaps in another function. Afaict all the old tests use `nil` for the last argument. If we have tests for lwt at the framer level they can apply the modification accordingly.

I am unsure why changing the framer constructor is bad in itself other than changing upstream code.

Reply:
It makes this patch unnecessarily big and you need to change all existing usages for no good reason.
In other words this increases chances of conflicts with upstream changes.

I suggest to add a new constructor newFramerWithExts or use a plain setter that needs to be called before one starts using the framer.

Reply:
> It makes this patch unnecessarily big and you need to change all existing usages for no good reason.
> In other words this increases chances of conflicts with upstream changes.
> 
> I suggest to add a new constructor newFramerWithExts or use a plain setter that needs to be called before one starts using the framer.

Okay, I will adjust appropriately.